ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleDBTest.java

```diff
@@ -0,0 +1,145 @@
+package org.cru.godtools.articles.aem.db;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.Observer;
+import android.arch.persistence.room.Room;
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.cru.godtools.articles.aem.model.Article;
+import org.cru.godtools.articles.aem.model.Attachment;
+import org.cru.godtools.articles.aem.model.ManifestAssociation;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import timber.log.Timber;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ArticleDBTest implements LifecycleOwner {
+
+    private ArticleDao mArticleDao;
+    private AttachmentDao mAttachmentDao;
+    private ManifestAssociationDao mAssociationDao;
+    private ArticleRoomDatabase db;
+
+    private Observer<List<Article>> articleObserver;
+
+    @Before
+    public void createDb() {
+        Context context = InstrumentationRegistry.getTargetContext();
+        db = Room.inMemoryDatabaseBuilder(context,
+                ArticleRoomDatabase.class).allowMainThreadQueries().build();
+        mArticleDao = db.articleDao();
+        mAttachmentDao = db.attachmentDao();
+        mAssociationDao = db.manifestAssociationDao();
+
+    }
+
+    @After
+    public void closeDb() {
+        db.close();
+    }
+
+    @Test
+    public void useAppContext() {
```

Comment:
test methods should be named for the specific method/functionality they are testing. for example:
- `verifyGetAllArticles()`
- `testInsertArticle()`
- `testDeleteArticle()`

This helps keep tests focused on testing specific functionality and makes it easier to track down when certain functionality changes.